### DISCUSSION TITLE
Participants can impersonate everyone in comments, even after registering/logging in

### DISCUSSION THREAD

**Discussion by ltguillaume** [lukevella/rallly#896] [Ideas]
**Describe the bug**
People can comment using every name, even after they have (reported the availability and) registered.

**To Reproduce**
1. Go to a poll
2. Add a comment, using any name you want
3. Report availability, register account
4. Be redirected back to poll
5. Add a comment, again using any (other) name

**Expected behavior**
Participants should not be able to choose their name over and over again.

- Participant shouldn't be able to choose the name of an existing participant
- If not registered yet, participant should be able to choose a name once under comments, then they should not be able to change anymore
- If the participant registers _after_ already having commented, they should not be able to specify a name, but just the email address
- Changing the name via the "You" context menu should also change the name in the comments

**Desktop (please complete the following information):**
 - Version 3.1.0

**Comment by lukevella** [comment]
So to clarify, it works as expected right now. It's not a highly refined implementation for sure and it will be improved as part of a bigger update. Moving this to discussions.

**Comment by icsy7867** [comment]
Just wanted to give my 2 cents here as well.  It does seem to make sense to not have to enter an email address if you are already logged in.  I am looking for a system internally for a large organization (We cannot run cloud based solutions, we can only run on-prem services when able.)

IMO, if a user is logged in, and there name and email address is available, those fields should be automatically filled in, and then make those fields "disabled" so that the user can see the text boxes, but cannot edit them.

Oh and if a user exists, a guest user shouldnt be able to use their email address.
